### PR TITLE
fix: strip <br> tags from textual error messages

### PR DESCRIPTION
**Pull request by pselkirk**
This is a minor follow-on to PR #5623.

As Jennifer noted, there are hard-coded `<br>` tags in the cutoff warning message. This converts them to spaces for API submissions, for a slight cleaner look.

There may be other html that we might want to sanitize, but this is the least-effort, least-impact change.

### CODE REVIEW

**Review by larseggert — COMMENTED** [review]
Might be a better idea to use `bleach` to deal with tag formatting variances?

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
This will improve the messages output through the API. I'm a little hesitant about the change because it's something of a half measure that still leaves some artifacts (extra spaces here and there). I'd really like to see this whole thing refactored to avoid assuming HTML all the way down in the stack and then trying to recover later.

Regardless of that, I think the mark_safe() is probably needed - I think that's worth pushing down to where the strings are constructed as I noted inline.

File: ietf/submit/forms.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def cleanup():  # called when context exited, even in case of exception
 
     def clean(self):
         if self.shutdown and not has_role(self.request.user, "Secretariat"):
-            raise forms.ValidationError(self.cutoff_warning)
+            raise forms.ValidationError(mark_safe(self.cutoff_warning))
```
[the remark this answers is not in the document]
Comment (pselkirk) [reply ietf/submit/forms.py:197]: Not sure exactly what you're asking for, but here's my view from the bottom of the rabbit hole (from the bottom up):

- I've been brought up to prefer `.format()` to `%` string interpolation, so I think I'm with you there.

- Aside from `Meeting.get_current_meeting()`, it looks like the arguments to the various forms of `self.cutoff_warning` are instances of `datetime.strftime("%Y-%m-%d %H:%M %Z")`, so I'm not sure what needs to be `mark_safe`'d before or after `format_html()`.

- Tangentially (while we're rewriting the underlying code), I'd be tempted to replace `set_cutoff_warnings()` with something like the following: 
```
    @property
    def cutoff_warning(self):
        if self._cutoff_warning is None:
            self._cutoff_warning = ""
            # ... contents of set_cutoff_warnings() ...
        return mark_safe(self._cutoff_warning)
```

- WRT the outer comment, I knew you'd spot the extra spaces. 🙂

File: ietf/submit/forms.py [hunk 2]
```diff
@@ -633,6 +633,12 @@ def __init__(self, request, *args, **kwargs):
         self.formats = ['xml', ]
         self.base_formats = ['xml', ]
 
+    def clean(self):
```
[the remark this answers is not in the document]
Comment (pselkirk) [reply ietf/submit/forms.py:636]: I'm not sure how that works. `upload_submission.html` references `form.cutoff_warning` as a django variable, so if there are going to be `<br>` tags (arguable, but above my pay grade), they need to already be in the string (unless we do something fancy with `@property`, as noted below).

**Comment by rjsparks** [comment]
I'm going to close this without taking it in. We can live with the `<br>` in the json respones for now.
We'll come back to this with a more comprehensive effort identifying places like this where html markup should be avoided or replaced with markdown.

**Pull request closed without merging**